Makefile

```diff
@@ -105,3 +106,9 @@ deploy: examples/admission-cert.pem
 		|sed "s/__admission_cert__/$(shell cat examples/admission-cert.pem|base64 -w0)/g" \
 		|sed "s/__admission_key__/$(shell cat examples/admission-key.pem|base64 -w0)/g" \
 		|kubectl apply -f -
+
+check-dependencies:
+	# We need mercurial for bitbucket.org/ww/goautoneg, otherwise dep hangs forever
+	which hg >/dev/null 2>&1 || apt update && apt install -y mercurial
+	which dep >/dev/null 2>&1 || (export DEP_RELEASE_TAG=v0.5.0; curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh)
```

Comment:
@xmudrii wdyt about using this method for https://github.com/kubernetes-sigs/cluster-api-provider-digitalocean/pull/120#discussion_r232020500?

Reply:
Checked with Marko on Slack, we can't use this for `dep` because using `which dep` requires `dep` to be in the `$PATH`. `dep` is installed in `$GOPATH/bin` but it is not in `$PATH`, while using prow.

We can use this for `hg` because it is installed system-wide and is in `$PATH`.

Updated so that it can work with prow.